wallet/src/repository/wallet/mod.rs

```diff
@@ -330,25 +321,68 @@ where
         Ok(())
     }
 
+    /// Filter transactions in a block (received from a node) if they belong to wallet accounts.
+    pub fn filter_wallet_transactions(
+        &self,
+        txns: &[types::Transaction],
+    ) -> Result<Vec<types::Transaction>> {
+        let state = self.state.read()?;
+
+        let mut filtered_txns = vec![];
+        for txn in txns {
+            // Inputs and outputs from different transaction types
+            let (inputs, outputs) = match txn {
+                types::Transaction::ValueTransfer(vt) => {
+                    (vt.body.inputs.clone(), vt.body.outputs.clone())
```

Comment:
Maybe you can remove this clones by using references?

```suggestion
                    (&vt.body.inputs, &vt.body.outputs)
```

Reply:
Tried but seems not possible.